onnxruntime/core/providers/cuda/gpu_data_transfer.cc

```diff
@@ -8,13 +8,12 @@ namespace onnxruntime {
 GPUDataTransfer::GPUDataTransfer() {
   // create streams, default is nullptr
   streams_[kCudaStreamDefault] = nullptr;
-  CUDA_CALL_THROW(cudaStreamCreateWithFlags(&streams_[kCudaStreamCopyIn], cudaStreamNonBlocking));
-  CUDA_CALL_THROW(cudaStreamCreateWithFlags(&streams_[kCudaStreamCopyOut], cudaStreamNonBlocking));
+  // put all copy stream back to default stream to avoid async memory conflict.
+  streams_[kCudaStreamCopyIn] = nullptr;
+  streams_[kCudaStreamCopyOut] = nullptr;
```

Comment:
should also modify the fence related code.  for e.g. in allocation_planner.cc we add fence based on the execution id.
if we only have a single stream, we can remove kCudaStreamCopyIn, kCudaStreamCopyIn, execution id, also fence. For copytohost, we can directly put the blocking inside the copytensor code I think.

Reply:
Seems too heavy fix for this. Will close this.